ElementX/Sources/Screens/RoomScreen/View/Timeline/GroupRoomTimelineView.swift

```diff
@@ -0,0 +1,72 @@
+//
+// Copyright 2023 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import SwiftUI
+
+struct GroupRoomTimelineView: View {
+    private let timelineItem: GroupTimelineItem
+    private let timelineViews: [RoomTimelineViewProvider]
+    
+    @State private var isExpanded = false
+
+    init(timelineItem: GroupTimelineItem) {
+        self.timelineItem = timelineItem
+        timelineViews = timelineItem.items.map { RoomTimelineViewProvider(timelineItem: $0) }
+    }
+    
+    var body: some View {
+        DisclosureGroup(ElementL10n.roomTimelineStateChanges(timelineItem.items.count),
+                        isExpanded: $isExpanded) {
+            ForEach(timelineViews) { timelineView in
+                timelineView.body
+            }
+        }
+        .disclosureGroupStyle(CustomDisclosureGroupStyle<Text>())
+        .animation(.noAnimation, value: timelineItem)
+    }
+}
+
+struct GroupRoomTimelineView_Previews: PreviewProvider {
+    static var previews: some View {
+        let item = GroupTimelineItem(items: [SeparatorRoomTimelineItem(text: "This is a separator"),
+                                             SeparatorRoomTimelineItem(text: "This is another separator")])
+        GroupRoomTimelineView(timelineItem: item)
+    }
+}
+
+struct CustomDisclosureGroupStyle<Label: View>: DisclosureGroupStyle {
```

Comment:
I'm guessing this needed to be generic in a previous implementation?
```suggestion
struct CustomDisclosureGroupStyle: DisclosureGroupStyle {
```

Reply:
Also, needs a more descriptive name :)

Reply:
I went with `CollapsibleRoomTimelineItemDisclosureGroupStyle` ¯\_(ツ)_/¯